advanced/matrix/matrix_yaniv.py

```diff
@@ -0,0 +1,162 @@
+            unity_list.append(tuple(sublist))
+        self.matrix_value = tuple(unity_list)
+        return self
+
+    def multiply(self, x):
+
+        a = list(self.matrix_value)
+        b = list(x.get_tuple())
+
+        matrix_after_multiply = []
+        for i in a:
+            i = list(i)
+            matrix_vector = []
+            for p in range(0, len(a), 1):
+                m = 0
+                num = 0
+                for j in b:
+                    j = list(j)
+                    num = num + (i[m] * j[p])
+                    m = m + 1
+                matrix_vector.append(num)
+            matrix_after_multiply.append(tuple(matrix_vector))
+        return matrix(tuple(matrix_after_multiply))
+
+    def __add__(self, x):
+        a = list(self.matrix_value)
+        b = list(x.get_tuple())
+
+        matrix_after_addition = []
+        for i in range(0, len(a), 1):
+            vector_matrix = []
+            for p in range(0, len(a[i]), 1):
+                vector_matrix.append(a[i][p] + b[i][p])
+            matrix_after_addition.append(tuple(vector_matrix))
+        return matrix(tuple(matrix_after_addition))
+
+    def __sub__(self, x):
+        a = list(self.matrix_value)
+        b = list(x.get_tuple())
+
+        matrix_after_substraction = []
+        for i in range(0, len(a), 1):
+            vector_matrix = []
+            for p in range(0, len(a[i]), 1):
+                vector_matrix.append(a[i][p] - b[i][p])
+            matrix_after_substraction.append(tuple(vector_matrix))
+        return matrix(tuple(matrix_after_substraction))
+
+    def multiply_scalar(self, x):
+        a = list(self.matrix_value)
+        b = int(x)
+
+        matrix_after_multiply = []
+
+        for i in a:
+            vector_matrix = []
+            for p in range(0, len(i), 1):
+                vector_matrix.append(i[p] * x)
+            matrix_after_multiply.append(tuple(vector_matrix))
+        return matrix(tuple(matrix_after_multiply))
+
+    def __truediv__(self, x):
+        a = list(self.matrix_value)
+        b = int(x)
+
+        matrix_after_divide = []
+
+        for i in a:
+            vector_matrix = []
+            for p in range(0, len(i), 1):
+                vector_matrix.append(i[p] / b)
+            matrix_after_divide.append(tuple(vector_matrix))
+        return matrix(tuple(matrix_after_divide))
+
+    @property
+    def tuples(self):
+        return self.matrix_value
+
+    def __ne__(self, x):
+        a = list(self.matrix_value)
+        b = list(x.get_tuple())
+
+        for i in range(0, len(a), 1):
+            for p in range(0, len(a[i]), 1):
+                if a[i][p] != b[i][p]:
+                    return True
+
+        return False
+
+    def __eq__(self, x):
+        a = list(self.matrix_value)
+        b = list(x.get_tuple())
+
+        for i in range(0, len(a), 1):
+            for p in range(0, len(a[i]), 1):
+                if a[i][p] != b[i][p]:
+                    return False
```

Comment:
There is a trick for that, you write one of them, and for the other you write ;
def ___ne__(self, x):
    return not self.__eq__(x)

Results in shorter and simpier code!